ddtrace/tracer/rules_sampler.go

```diff
@@ -484,6 +484,10 @@ func (rs *traceRulesSampler) applyRate(span *Span, rate float64, now time.Time,
 	span.mu.Lock()
 	defer span.mu.Unlock()
 
+	if span.finished {
```

Comment:
nit: most (all?) of these finished checks currently come with the same comment which I think we should continue duplicating until we have a better static analysis solution.

see https://github.com/DataDog/dd-trace-go/issues/3472#issuecomment-2839599603 for the comment

Reply:
Done in [a9683c5](https://github.com/DataDog/dd-trace-go/pull/3474/commits/a9683c5525029ed01a416053472409a91656aa6a)